staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -288,17 +307,18 @@ type Backoff struct {
 	Cap time.Duration
 }
 
-// Step (1) returns an amount of time to sleep determined by the
-// original Duration and Jitter and (2) mutates the provided Backoff
-// to update its Steps and Duration.
+// Step returns an amount of time to sleep determined by the
+// original Duration and Jitter. The backoff is mutated to update its
+// Steps and Duration.
 func (b *Backoff) Step() time.Duration {
 	if b.Steps < 1 {
 		if b.Jitter > 0 {
 			return Jitter(b.Duration, b.Jitter)
 		}
 		return b.Duration
+	} else {
+		b.Steps--
```

Comment:
this is not really needed, right? the previous if always return
I think this is more confusing , since next instructions seems to follow a different flow than b.Step

Reply:
The else?  Yeah, should be removed now.